lib/babylon-to-espree/convertAST.js

```diff
@@ -31,16 +33,15 @@ const astTransformVisitor = {
     node._babelType = node.type;
 
     if (node.innerComments) {
-      node.trailingComments = node.innerComments;
       delete node.innerComments;
     }
 
     if (node.trailingComments) {
-      convertComments(node.trailingComments);
+      delete node.trailingComments;
```

Comment:
Just for my own understanding, does ESLint just not used these properties?

Reply:
That's right! We dropped the need for comment attachment to happen at the parser level in v4 (see docs [here](https://github.com/eslint/eslint/blob/caeb223c4f7b0b6fe35e5348ae0df4c6446b5bed/docs/user-guide/migrating-to-4.0.0.md#-ast-nodes-no-longer-have-comment-properties)). We decided to move away from thinking about comments in the context of AST nodes (which had to be reimplemented by every parser and was buggy at best) and instead to just think of them in the context of tokens. ESLint now just takes the tokens and comments arrays provided by the parser and gives rules access to comments through the use of utility methods like `sourceCode#getTokenBefore(nodeOrToken, { includeComments: true })`.

Reply:
I was talking to Kai about being able to turn off even adding the comments in @babel/parser but if the only use case is for babel-eslint we probably don't need to make that option then? Unless there's a case for it being much faster (would need to test)

Reply:
I could explore that - I haven't looked at how comment attachment works in @babel/parser in a long time now, but if it's an easy thing to add I'm happy to do it! That said, I don't think it needs to be tied to this release, since it won't result in any user-facing changes.